pkg/gui/types/common.go

```diff
@@ -196,9 +201,9 @@ type MenuItem struct {
 	Tooltip string
 
 	// Can be used to group menu items into sections with headers. MenuItems
-	// with the same Section string should be contiguous, and will automatically
+	// with the same Section title should be contiguous, and will automatically
 	// get a section header.
-	Section string
+	Section MenuSection
```

Comment:
This is a bit redundant; all menu items in a section have a column, but only the first one will be used. Alternatively, we could pass in the column to align to via `CreateMenuOptions`, but then we are restricted to aligning all section headers to the same column. With the current design we could theoretically align different section headers to different columns.

Reply:
The solution of course is to use a pointer to a section here, and use pointer comparison to tell whether two items are in the same section. Much cleaner, I added another two fixups for that.

(The fixups are starting to get a little out of hand, let me know when it's a good time to squash them.)

Reply:
Feel free to squash now